server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -1719,6 +1720,99 @@ def test_post_publish_repo_missing_distributor(self, mock_repo_qs, mock_factory)
         self.assertTrue(response.error_code is error_codes.PLP0016)
 
 
+class TestRepoDownload(unittest.TestCase):
+    """Tests for RepoDownload."""
+
+    @mock.patch('pulp.server.webservices.views.decorators._verify_auth',
+                new=assert_auth_EXECUTE())
+    @mock.patch('pulp.server.webservices.views.repositories.content_controller')
+    @mock.patch('pulp.server.webservices.views.repositories.model.Repository.objects')
+    def test_post_download_repo(self, mock_repo_qs, mock_content_controller):
+        """Test that a repo download task is dispatched."""
+        # Setup
+        download_repo = RepoDownload()
+
+        # Tests
+        try:
```

Comment:
You can use ``assertRaises`` as a context manager, which would improve readability. You probably need to use ``unittest2`` for that, in which case you can just import ``unittest`` from our ``compat`` module. It goes something like:

```
with self.assertRaises(exception.OperationPostponed) as exception:
  do_thing()

# now make assertions about the exception
self.assertEqual(exception.http_status_code, 202)
```